Lib/uuid.py

```diff
@@ -581,6 +589,18 @@ def _netbios_getnode():
     _generate_time_safe = getattr(_uuid, "generate_time_safe", None)
     _UuidCreate = getattr(_uuid, "UuidCreate", None)
     _has_uuid_generate_time_safe = _uuid.has_uuid_generate_time_safe
+
+    if sys.platform == "darwin" and _generate_time_safe is not None:
+        # On macOS check if generate_time_safe uses the MAC adddress
+        # of the Touch Bar on Intel Macbooks and ignore the function
+        # if it does because this is a single MAC adress for all 
+        # devices. See GH-85724
+        _x = _generate_time_safe()[0]
+        if _x.endswith(bytes.fromhex(_MACOS_TOUCHBAR_MAC.replace(":", ""))):
+            _has_uuid_generate_time_safe = False
+            _generate_time_safe = None
```

Comment:
Would it make sense/be possible to implement this test directly in the _uuid module: remove the function if it returns this address?

Reply:
Hmmm... Possible yes, sensible maybe not.

The implementation of ``_uuid.generate_time_safe`` has 3 or 4 variants (depending on how you count) and because of that an implementation in C mirrors the Python code above but is more verbose and likely not faster than the Python implementation. 

I have a patch, but haven't tested it on a device with a Touch Bar yet. That patch adds 49 lines of code to _uuidmodule.c and removes the 4 lines of python code above.